measures/forms.py

```
@@ -692,10 +689,22 @@ def save(self, commit=True):
                 "component_measure",
             )
 
+        # Footnotes added via "Add another footnote" button
         footnote_pks = [
-            dct["footnote"]
-            for dct in self.request.session.get(f"formset_initial_{sid}", [])
+            form["footnote"]
+            for form in self.request.session.get(f"formset_initial_{sid}", [])
         ]
+
+        # Footnote submitted directly via "Save" button
+        if hasattr(self.request, "POST"):
```

Are there circumstances when this `save()` is called from something other than servicing a POST request?

---

If not, then it may be preferable to let things blow up on invalid access to `self.request.POST`, avoiding silent failures.

---

There were a couple form tests that called the form's `save()` method without a POST request, but I've updated them and removed this conditional.